.ci/setenvconfig

```diff
@@ -296,6 +298,7 @@ MONITORING_SECRETS=${PROJECT_PATH}/.ci/monitoring-secrets.json
 
 PIPELINE=e2e/main
 BUILD_NUMBER=$BUILD_NUMBER
+JOB_NAME=$JOB_NAME
```

Comment:
IIUC, we look at `JOB_NAME` to know if we need retrive the gcp creds (for the bucket) and know if we can run the diagnostic.
Wouldn't `E2E_FAILFAST=true` be more explicit?

Reply:
To your point, wouldn't `CI` environment variable work even better? If running in CI, we can run the diagnostics on each test failure...thoughts?

Reply:
oh second thought, idk. Job name is used to build the namespace for the bucket upload as well:
```
	run(ctx, "gsutil", "cp", "/tmp/*.zip", fmt.Sprintf("gs://%s/jobs/%s/%s/", testCtx.GSBucketName, testCtx.JobName, testCtx.BuildNumber))
```
I don't know if I think `E2E_FAILFAST` is more clear, as we are never failing fast at this point, right?

Reply:
> wouldn't CI environment variable work even better?

I thought we wanted to control whether we fail fast or not. If the goal is only to switch the feature in CI, yes, I think `CI` is a good candidate.

>  Job name is used to build the namespace for the bucket upload as well

We may not need it and use `CLUSTER_NAME`, see https://github.com/elastic/cloud-on-k8s/pull/6363#discussion_r1139367438.

Reply:
> I thought we wanted to control whether we fail fast or not. If the goal is only to switch the feature in CI, yes, I think `CI` is a good candidate.
> 
> > Job name is used to build the namespace for the bucket upload as well
> 
> We may not need it and use `CLUSTER_NAME`, see [#6363 (comment)](https://github.com/elastic/cloud-on-k8s/pull/6363#discussion_r1139367438).

I just re-read the issue. There's no mention of toggling on/off. I'll investigate and get back to you

Reply:
There is no need to toggle fail fast behaviour on and off IMO. Thus there is also no need for an environment variable with that name. 

The only distinction we need to introduce is between local runs and CI runs with access to the relevant bucket and where we know that eck-diagnostics is in scope and we want to execute it after failures. Whether we want to use `CI` or `JOB_NAME`  depends mostly if one of them implies the others. E.g if `CI` implies that `JOB_NAME` is set then  `CI` is more expressive as the main variable that control the different behaviour.